include/sentry.h

```diff
@@ -1226,6 +1226,57 @@ SENTRY_EXPERIMENTAL_API void sentry_options_set_traces_sample_rate(
 SENTRY_EXPERIMENTAL_API double sentry_options_get_traces_sample_rate(
     sentry_options_t *opts);
 
+/* -- Performance Monitoring/Tracing APIs -- */
+
+/**
+ * Constructs a new transaction context to be passed
+ * into `sentry_start_transaction`.
+ *
+ * See
+ * https://docs.sentry.io/platforms/native/enriching-events/transaction-name/
+ * for an explanation of a transaction's `name`, and
+ * https://develop.sentry.dev/sdk/performance/span-operations/ for conventions
+ * around an operation's name.
+ *
+ * Also see https://develop.sentry.dev/sdk/event-payloads/transaction/#anatomy
+ * for an explanation of `operation`, in addition to other properties and
+ * actions that can be performed with a transaction.
+ */
+SENTRY_EXPERIMENTAL_API sentry_value_t sentry_value_new_transaction_context(
+    const char *name, const char *operation);
+
+/**
+ * Sets the name of a transaction on a `transaction_context`.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_transaction_context_set_name(
+    sentry_value_t transaction_context, const char *name);
+
+/**
+ * Sets the operation of a transaction on a `transaction_context`.
+ *
+ * See https://develop.sentry.dev/sdk/performance/span-operations/ for
+ * conventions on `operation`s.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_transaction_context_set_operation(
+    sentry_value_t transaction_context, const char *operation);
+
+/**
+ * Sets the sampled field on a `transaction_context`. When turned on, the
+ * transaction constructed from the `transaction_context` will bypass all
+ * sampling options and always be sent to sentry. If this is explicitly turned
+ * off in the `transaction_context`, the transaction will never be sent to
+ * sentry.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_transaction_context_set_sampled(
+    sentry_value_t transaction_context, int sampled);
+
+/**
+ * Removes the sampled field on a `transaction_context`. The transaction
+ * contructed from it will use the sampling rate as defined in `sentry_options`.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_transaction_context_remove_sampled(
+    sentry_value_t transaction_context);
```

Comment:
i'm strongly considering renaming all of these and stripping the `_context` bit of this. instead, the API would most likely look more like this:

- `sentry_value_new_transaction`
- `sentry_transaction_set_name`
- `sentry_transaction_set_operation`
- `sentry_transaction_set_sampled`
- `sentry_start_transaction`
- `sentry_transaction_finish`

it does mean that we paper over the difference between an inert transaction and one that is active, but you could argue that most objects are the same way (see Exceptions, Threads, Messages, etc)

Reply:
this is still relevant despite the outdated tag.

Reply:
I'd be ok with your newly proposed names as well

Reply:
`sentry_start_transaction` kind of sticks out among those name tbh. Should that be renamed to `sentry_transaction_start`?

Reply:
let me get some opinions from some of the people working on perfmon to see if they're averse to us changing the name a little in native - if they're adamant about having a function named `start_transaction`, perhaps we can just offer both options (`sentry_start_transaction` and `sentry_transaction_start`) and let users decide?